controllers/volumereplication_controller.go

```diff
@@ -205,33 +208,97 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 }
 
 // markVolumeAsPrimary defines and runs a set of tasks required to mark a volume as primary
-func (r *VolumeReplicationReconciler) markVolumeAsPrimary(volumeID string, parameters, secrets map[string]string) error {
+func (r *VolumeReplicationReconciler) markVolumeAsPrimary(volumeReplicationObjectRef types.NamespacedName, volumeID string, parameters, secrets map[string]string) error {
 	c := replication.CommonRequestParameters{
 		VolumeID:    volumeID,
 		Parameters:  parameters,
 		Secrets:     secrets,
 		Replication: r.Replication,
 	}
-	force := false
-	var markVolumeAsPrimaryTasks = []*tasks.TaskSpec{
-		{
-			Name: enableVolumeReplication,
-			Task: replication.NewEnableTask(c),
-		},
-		{
-			Name: promoteVolume,
-			Task: replication.NewPromoteVolumeTask(c, force),
-		},
+
+	instance := replicationv1alpha1.VolumeReplication{}
+	err := r.Client.Get(context.TODO(), volumeReplicationObjectRef, &instance)
```

Comment:
do we need to get the instance again? just use the existing instance we already got?

Reply:
Using existing instance.